### PR TITLE
Update OpenTelemetry to 1.31.0

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseNetworkHelper.java [hunk 1]
```diff
@@ -35,20 +36,20 @@ HttpRequest buildPingRequest(
...
-        request.setHeader(QPS_MACHINE_NAME_HEADER, machineName);
-        request.setHeader(QPS_STREAM_ID_HEADER, quickPulseId);
-        request.setHeader(QPS_INSTANCE_NAME_HEADER, instanceName);
+        request.setHeader(HttpHeaderName.fromString(QPS_ROLE_NAME_HEADER), roleName);
+        request.setHeader(HttpHeaderName.fromString(QPS_MACHINE_NAME_HEADER), machineName);
+        request.setHeader(HttpHeaderName.fromString(QPS_STREAM_ID_HEADER), quickPulseId);
+        request.setHeader(HttpHeaderName.fromString(QPS_INSTANCE_NAME_HEADER), instanceName);
         request.setHeader(
-            QPS_INVARIANT_VERSION_HEADER, Integer.toString(QuickPulse.QP_INVARIANT_VERSION));
+            HttpHeaderName.fromString(QPS_INVARIANT_VERSION_HEADER), Integer.toString(QuickPulse.QP_INVARIANT_VERSION));
```
Comment (trask) [inline sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseNetworkHelper.java:44]: i didn't follow how these changes are related to OpenTelemetry update?

**Review by trask — APPROVED** [review]

File: sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseNetworkHelper.java (hunk 1 above)
Comment (heyams) [reply sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseNetworkHelper.java:44]: deprecated warning

File: sdk/core/azure-core-metrics-opentelemetry/pom.xml [hunk 2]
```diff
@@ -136,11 +136,11 @@
...
-                <include>io.opentelemetry:opentelemetry-exporter-otlp:[1.28.0]</include> <!-- {x-include-update;io.opentelemetry:opentelemetry-exporter-otlp;external_dependency} -->
+                <include>io.opentelemetry:opentelemetry-api:[1.31.0]</include> <!-- {x-include-update;io.opentelemetry:opentelemetry-api;external_dependency} -->
+                <include>io.opentelemetry:opentelemetry-sdk:[1.31.0]</include> <!-- {x-include-update;io.opentelemetry:opentelemetry-sdk;external_dependency} -->
+                <include>io.opentelemetry:opentelemetry-sdk-testing:[1.31.0]</include> <!-- {x-include-update;io.opentelemetry:opentelemetry-sdk-testing;external_dependency} -->
+                <include>io.opentelemetry:opentelemetry-exporter-logging:[1.31.0]</include> <!-- {x-include-update;io.opentelemetry:opentelemetry-exporter-logging;external_dependency} -->
+                <include>io.opentelemetry:opentelemetry-exporter-otlp:[1.31.0]</include> <!-- {x-include-update;io.opentelemetry:opentelemetry-exporter-otlp;external_dependency} -->
```
Comment (alzimmermsft) [inline sdk/core/azure-core-metrics-opentelemetry/pom.xml:143]: I believe these can be removed as these dependencies are only used in `<scope>test</scope>` and the `bannedDependencies` only checks runtime dependencies.

Comment (heyams) [reply sdk/core/azure-core-metrics-opentelemetry/pom.xml:143]: i will leave its owner to clean it up. Some are for testing, and some are not.

**Pull request merged**